pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -261,5 +266,138 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 		log.Info(fmt.Sprintf("%s successful ", comment))
 		return reconcile.Result{}, nil
 	}
+
+	// data ingress logic
+	// for every ingress on the cluster
+	// if it is of type default=true check if [listening && dnsName && certificate] matches with publishingStrategy CR's default application ingress
+	// if it matches, exit reconcile
+	// if it does not match, delete on-cluster default ingress and replace it with new one
+	// else if default=false
+	// for every default=false on publishing CR's applicationIngress
+	// check if publishing CR's applicationIngress [listening && dnsName && certificate] matches with on-cluster's ingress
+	// create new ingress if does not match
+
+	// get a list of all ingress on the cluster
+	ingressControllerList := &operatorv1.IngressControllerList{}
+	listOptions := []client.ListOption{
+		client.InNamespace("openshift-ingress-operator"),
+	}
+	err = r.client.List(context.TODO(), ingressControllerList, listOptions...)
+	if err != nil {
+		log.Error(err, "Cannot get list of ingresscontroller")
+		return reconcile.Result{}, err
+	}
+
+	// for every ingress on the cluster
+	// if it is default cehck if scope, listening, certificate matches with cr app ingress
+
+	// For each app ingress
+	// For each item in the ps.applicationIngress
+	// Check if the appingress definition is the same as the item in the list
+	// If none of them match, do
+
+	for _, publishingStrategyIngress := range instance.Spec.ApplicationIngress {
+		for _, ingressController := range ingressControllerList.Items {
+			if doesIngressMatch(&publishingStrategyIngress, &ingressController) {
+				log.Info("match, move on to next cycle")
+				continue
+			} else {
+				// create a LocalObjectReference and fill it out secret passed in by OCM
+				newCertificate := &corev1.LocalObjectReference{
+					Name: publishingStrategyIngress.Certificate.Name,
+				}
+				if ingressController.Name == "default" {
+					// delete existing default ingresscontroller
+					err := r.client.Delete(context.TODO(), &ingressController)
+					if err != nil {
+						log.Error(err, "failed to delete existing ingresscontroller")
+						return reconcile.Result{}, err
+					}
+
+					// create a new ingresscontroller
+					newIngressControllerCR, err := newApplicationIngressControllerCR("default", string(publishingStrategyIngress.Listening), publishingStrategyIngress.DNSName, newCertificate)
+					if err != nil {
+						log.Error(err, "failed to generate data for new ingresscontroller CR")
```

Comment:
Here we find an error but we don't return and still continue to next step potentially using a bogus `newIngressControllerCR` variable

Reply:
return reconcile.Result{}, err added